src/redux/slices/events/middleware.ts

```diff
@@ -51,14 +56,62 @@ const getSingleEvent = (id: string) => async (dispatch: AppDispatch) => {
   }
 }
 
+const createEvent = (formData: ICreateEventParams) => async (dispatch: AppDispatch) => {
+  try {
+    dispatch(setCreateEventSubmitSuccess(true))
+    await API.events.createEvent(formData)
+  } catch (error) {
+    /* empty */
+  }
+}
+
 const clearEventsList = () => async (dispatch: AppDispatch) => {
   dispatch(setEventsList([]))
   dispatch(setPageSize(0))
   dispatch(setTotalItems(0))
 }
 
+const resetCreateEventsSubmitSuccess = () => (dispatch: AppDispatch) => {
+  dispatch(setCreateEventSubmitSuccess(false))
+}
+
+const deleteIndividualEvent = (uuid: string) => async (dispatch: AppDispatch) => {
+  try {
+    dispatch(setDeleteEventLoading(true))
+
+    await API.events.deleteIndividualEvent(uuid)
+
+    dispatch(clearEventsList())
+
+    dispatch(eventsMiddleware.getEventsList(1))
+  } catch (error) {
+    /* empty */
+  } finally {
+    dispatch(setDeleteEventLoading(true))
+  }
+}
+
+const changeEvent = (formData: IChangeEventFormProps) => async (dispatch: AppDispatch) => {
+  try {
+    await API.events.changeEvent(formData)
+
+    dispatch(setChangeEventSubmitSuccess(true))
+  } catch (error) {
+    /* empty */
+  }
+}
```

Comment:
Why no loading?

Reply:
In that case let's set redirect logic here instead of keeping state for that

Reply:
done